File: apps/android/app/src/main/java/com/example/creditcard/utils/CardAttachmentPolicy.kt
package com.example.creditcard.utils

/** Limits apply only when adding new images, never while reading an existing card or snapshot. */
object CardAttachmentPolicy {
    const val MAX_IMAGES = 12
    const val MAX_SOURCE_BYTES = 10 * 1024 * 1024
    const val MAX_EDGE = 1600
    fun canAppend(existingCount: Int, additionalCount: Int) = existingCount >= 0 && additionalCount >= 0 &&
        existingCount <= MAX_IMAGES && additionalCount <= MAX_IMAGES - existingCount
}

File: apps/macos/Domain/CardAttachmentPolicy.swift
import Foundation

/// Limits apply to NEW local imports only; old/remote attachments are never truncated.
public enum CardAttachmentPolicy {
    public static let maximumImages = 12
    public static let maximumSourceBytes = 10 * 1024 * 1024
    public static let maximumEdge = 1600
    public static func canAppend(existingCount: Int, additionalCount: Int) -> Bool {
        existingCount >= 0 && additionalCount >= 0 && existingCount <= maximumImages && additionalCount <= maximumImages - existingCount
    }
}
